### PR TITLE
[DRAFT] Introduce one-build

### PR DESCRIPTION
**Pull request by mhs4670go**
This draft is for one-build.

You can run `one-import-tf` with this draft as of now. Please review the configuration file.
```bash
$  ./one-build -c one-build.cfg
```
Once the configuration specification is approved, I'll write the rest of `one-build`.
Related : #4278 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
A. I think we need to consider for upcoming tasks:
- import TF saved model, keras model: no need of input/output arrays and shapes
- import ONNX: also no need of input/output arrays and shapes

but, I think there's no critical issues for these
please simulate about what changes are needed to the codes

B. Using flat `.cfg`
- not organized/structured so it's a bit hard to distinguish arguments that are depend on
   - we may use comments to separate visually
   - not sure this will be good for users and problem solving in the future.
- as @lemmaa suggested using `INI` can solve this issue; or using json format.

File: compiler/one-cmds/one-build [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+# test if getopt is enhanced version
+getopt --test > /dev/null
+if [ $? -ne 4 ]; then
+  echo "[ERROR] Your system doesn't have enhanced getopt"
+  exit 2
+fi
+
+# usage
+function Usage()
```
Comment (glistening) [inline compiler/one-cmds/one-build:27]: @mhs4670go (optional) without `function`, it become more short and portable.

**Comment by hyunsik-yoon** [comment]
It would be much better if `-h` option shows _what this script is for_ (purpose of this script). :-)
(At a glance of this PR title, I thought that it is some kinda of build script for ONE (compiler + runtime) for CI and if `-h` provides such info, it will be better. :-)

**Comment by mhs4670go** [comment]
@hyunsik-yoon Well, I think so. We can rename it to `one-compile`.

**Comment by seanshpark** [comment]
Leaving an idea from short off-line discussion

*How about providing several templates like `one-build.cfg` for each format?*
- There are items that must be filled and others left alone (don't care)
   - users may not know which item to fill
- Instead, we provide template of cfg for each formats(TF, BCQ, TFLITE and later TF/SavedModel, TF/Keras, ONNX)
   - users should fill for each items
   - but, still there are lots of common (optimize, quantize... ) optionals

**Comment by seanshpark** [comment]
If may confuse by the this PR title but this tool (one-build) will be installed as a command line.

**Comment by seanshpark** [comment]
> I thought that it is some kinda of build script for ONE (compiler + runtime) for CI 

@hyunsik-yoon , the file is `compiler/one-cmds/one-build` not in `tools/*` or such.

File: compiler/one-cmds/one-build.cfg [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+output_path="inception_v3.circle"
+
+# driver to use
+use_one_import_tf="true"
+use_one_import_bcq="false"
+use_one_import_tflite="false"
+use_one_optimize="true"
+use_one_quantize="false"
+use_one_pack="true"
+use_one_codegen="false"
```
Comment (mhs4670go) [inline compiler/one-cmds/one-build.cfg:21]: I reordered `use` options to make it more readable. Users first select what driver to use and then fill the rest of other options. If users miss some options, the script occurs an error.

File: compiler/one-cmds/one-build.cfg [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+### one-import-tf ### one-import-bcq ###
```
Comment (seanshpark) [inline compiler/one-cmds/one-build.cfg:23]: Would be be better if add a separate line like this?
```
# ----------------------------------------------------------------------------
### one-import-tf ### one-import-bcq ###
```

File: compiler/one-cmds/one-build.cfg [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+optimize_resolve_customop_batchmatmul="false"
+# resolve_customop_matmul       Enable ResolveCustomOpMatMulPass Pass
+optimize_resolve_customop_matmul="false"
+# optimize_output_path          (Optional) Path to intermediate output of one-optimize
+optimize_optimize_output_path=""
+
+### one-quantize ###
+# input_dtype                   Input data type (supported: float32, default=float32)
+quantize_input_dtype=""
+# quantized_dtype               Output quantized data type (supported: uint8, default=uint8)
```
Comment (seanshpark) [inline compiler/one-cmds/one-build.cfg:60]: We now surpport `int16`

File: compiler/one-cmds/one-build [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  echo "See \"one-build.cfg\" for more details."
+  echo ""
+  echo "[OPTIONS TO OVERWRITE]"
+  echo "Sometimes you want to change certain options without touching the configuration file."
+  echo "If you pass the option directly to the command line, the option is processed prior to "
+  echo "the configuration file. A list of option names can be found in \"one-build.cfg\"."
+}
+
+# version
+DRIVER_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```
Comment (seanshpark) [inline compiler/one-cmds/one-build:43]: (optional) maybe move `DRIVER_PATH` above `# version` ?

**Comment by mhs4670go** [comment]
@seanshpark You can use `one-build` as of now:) Maybe needs to do more tests.

**Comment by mhs4670go** [comment]
I'll post a new PR because `one-build` design will be changed a little bit.

**Pull request closed without merging**